makefiles/arch/msp430.inc.mk

```diff
@@ -17,9 +17,16 @@ LINKFLAGS += $(CFLAGS_CPU) $(CFLAGS_DBG) $(CFLAGS_OPT)
 LINKFLAGS += -Wl,--gc-sections -Wl,-L$(MSP430_SUPPORT_FILES)/include
 LINKFLAGS += -T $(MSP430_SUPPORT_FILES)/include/$(CPU_MODEL).ld
 LINKFLAGS += -T $(RIOTCPU)/msp430_common/ldscripts/msp430_common.ld
-LINKFLAGS += $(RIOTCPU)/msp430_common/ldscripts/xfa.ld
+LINKFLAGS += -T $(RIOTCPU)/msp430_common/ldscripts/xfa.ld
 
 OPTIONAL_CFLAGS_BLACKLIST += -fdiagnostics-color
 OPTIONAL_CFLAGS_BLACKLIST += -Wformat-overflow
 OPTIONAL_CFLAGS_BLACKLIST += -Wformat-truncation
 OPTIONAL_CFLAGS_BLACKLIST += -gz
+
+ifeq ($(TOOLCHAIN),gnu)
+  # https://gcc.gnu.org/bugzilla/show_bug.cgi?id=105523
+  ifneq ($(findstring 12.,$(shell $(TARGET_ARCH)-gcc --version 2>/dev/null)),)
```

Comment:
```suggestion
  ifeq (12, $(GCC_VERSION))
```

should also work

Reply:
Cool :) I should update the AVR hack then as well :)

Reply:
This doesn't work, as this is included prior to `makefiles/toolchain/gnu.inc.mk`. But now I wonder if we should just add `--param=min-pagesize=0` to GCC for all architectures when GCC is version 12.

Reply:
If this is not specific to MSP430 it should be in the common GCC makefile

Reply:
The documentation on the flag is super sparse. In https://gcc.gnu.org/pipermail/gcc-help/2023-March/142395.html it is assumed that it is the smallest legal address in the data address space (and the same interpretation is also stated in the issue). If that is true (which is kind of contradicting to the documentation and the name of the parameter), it would make sense to set this globally.